nodes/iol/iol.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Nokia
+// Licensed under the BSD 3-Clause License.
+// SPDX-License-Identifier: BSD-3-Clause
+
+package cisco_iol
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"regexp"
+
+	"github.com/srl-labs/containerlab/nodes"
+	"github.com/srl-labs/containerlab/types"
+	"github.com/srl-labs/containerlab/utils"
+)
+
+var (
+	kindnames          = []string{"cisco_iol"}
+	defaultCredentials = nodes.NewCredentials("admin", "admin")
+
+	InterfaceRegexp = regexp.MustCompile(`(?:e|Ethernet)\s?0/(?P<port>\d+)$`)
+	InterfaceOffset = 1
+	InterfaceHelp   = "e0/X or Ethernet0/X (where X >= 1) or ethX (where X >= 1)"
+)
+
+// Register registers the node in the NodeRegistry.
+func Register(r *nodes.NodeRegistry) {
+	r.Register(kindnames, func() nodes.Node {
+		return new(iol)
+	}, defaultCredentials)
+}
+
+type iol struct {
+	nodes.VRNode
+}
```

Comment:
@kaelemc somehow these iol images do not work for me.

This is what I get in the logs

```
❯ docker logs -f clab-iol-switch 
2024-10-03 10:42:12,883 INFO Set uid to user 0 succeeded
2024-10-03 10:42:12,887 INFO RPC interface 'supervisor' initialized
2024-10-03 10:42:12,887 INFO supervisord started with pid 1
2024-10-03 10:42:13,890 INFO spawned: 'entrypoint' with pid 7
2024-10-03 10:42:14,892 INFO success: entrypoint entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2024-10-03 10:42:14,892 INFO exited: entrypoint (exit status 0; expected)
2024-10-03 10:42:24,154 INFO spawned: 'iol' with pid 39
2024-10-03 10:42:24,156 INFO success: iol entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2024-10-03 10:42:24,222 INFO reaped unknown pid 8 (exit status 120)
```

I believe you don't have the process reaped as I do?

Reply:
also I noticed that interfaces of the container do not have v4 addresses
something you remove in the entrypoint?

```
root@router2:/opt/iol# ifconfig 
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 2001:172:20:20:42:acff:fe14:1405  prefixlen 64  scopeid 0x0<global>
        ether 02:42:ac:14:14:05  txqueuelen 0  (Ethernet)
        RX packets 359  bytes 39937 (39.0 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 102  bytes 13978 (13.6 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

Reply:
as discussed in discord:
- ip addresses are claimed by the IOL, hence they are not visible in the default netns